apps/meteor/client/sidebar/RoomMenu.tsx

```diff
@@ -171,19 +171,24 @@ const RoomMenu = ({
 		);
 	});
 
-	const handleToggleRead = useMutableCallback(async () => {
+	const handleToggleRead = useEffectEvent(async () => {
 		try {
 			if (isUnread) {
 				await readMessages({ rid, readThreads: true });
 				return;
 			}
-			await unreadMessages(undefined, rid);
+
 			if (subscription == null) {
 				return;
 			}
+
 			LegacyRoomManager.close(subscription.t + subscription.name);
 
-			router.navigate('/home');
+			if (roomOpen) {
+				router.navigate('/home');
+			}
```

Comment:
Is this related to the original issue? If not, let's create a specific PR for that